### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by yeoutzer**
Update developer guide.
Add some assertions.
Minor change to user guide.
Update some files to explicitly import classes instead of wildcard imports.

### CODE REVIEW

**Comment by zenlyj** [comment]
Looks good! Documentation is detailed and well written.

**Comment by linhns** [comment]
Nice job. Will merge when everyone has updated UG and DG

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -261,11 +284,60 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+![FilterSequenceDiagram](images/FilterSequenceDiagram.png)
+<div markdown="span" class="alert alert-info">
+
+:information_source: **Note:** The lifeline for `FilterCommandParser` and `FilterCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+</div>
 
-_{Explain here how the data archiving feature will be implemented}_
+The following activity diagram summarizes what happens when a user executes the filter command:
+![FilterActivityDiagram](images/FilterActivityDiagram.png)
```
Comment (vuminhhieunus2019) [inline docs/DeveloperGuide.md:340]: You can consider enlarge the image a little since I find the text in your sequence diagram is a bit small.

**Review by vuminhhieunus2019 — COMMENTED** [review]
Overall, nice work. Thanks for adding the table of content in the DG.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (yeoutzer) [reply docs/DeveloperGuide.md:340]: Yes, thank you for the suggestion. I have tried changing the sequence diagram accordingly.

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 2]
```diff
@@ -237,6 +242,6 @@ public void parseSortOptions_invalidOption_throwsParseException() {
     @Test
     public void parseSortOptions_validOption_returnsSortOptions() throws Exception {
         SortOptions expectedOption = SortOptions.PRIORITY_ASCENDING;
-        assertEquals(expectedOption, ParserUtil.parseSortOptions("priority", "a"));
+        assertEquals(expectedOption, parseSortOptions("priority", "a"));
```
Comment (linhns) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:245]: I think can just leave it in the old form as ParserUtil is static but looks fine

**Review by linhns — COMMENTED** [review]
Overall, LGTM

**Pull request merged**